modules/xphoto/test/test_denoise_bm3d.cpp

```diff
@@ -76,7 +76,7 @@ namespace opencv_test { namespace {
 
         // BM3D: two different calls doing exactly the same thing
         cv::Mat result, resultSec;
-        cv::xphoto::bm3dDenoising(original, cv::Mat(), resultSec, 10, 4, 16, 2500, 400, 8, 1, 0.0f, cv::NORM_L2, cv::xphoto::BM3D_STEPALL);
+        cv::xphoto::bm3dDenoising(original, result, resultSec, 10, 4, 16, 2500, 400, 8, 1, 0.0f, cv::NORM_L2, cv::xphoto::BM3D_STEPALL);
         cv::xphoto::bm3dDenoising(original, result, 10, 4, 16, 2500, 400, 8, 1, 0.0f, cv::NORM_L2, cv::xphoto::BM3D_STEPALL);
```

Comment:
You should not use "result" twice. It is not guaranteed that it is updated by second call.
Perhaps we should leave tests "as is" or replace `Mat()` to `noArray()`.